### ISSUE TITLE
Mismatching field names

### ISSUE THREAD

**Issue by strogonoff** [geolexica/isotc211-glossary#6]
Currently, certain field names do not match the format expected by the software that is used to explore and manage terminology, and in future to generate deliverables from terminology registries.

Data model is currently specified here as TypeScript types:

https://github.com/glossarist/glossarist-desktop/blob/master/src/models/concepts.ts

Example mismatch: normative status should be `normativeStatus`, not `normative_status`. (If that’s the only mismatch, then it should be quick to fix.)

I can see that the motivation was consistency, but it should be kept in mind that data repositories are currently used as software database, hence changes to data break the software.

1. Current policy is to use camelCase, per TypeScript convention for identifiers, except for legacy fields that use snake_case. This was not reflected much in the upstream database, since in development the software had been used with [a fork](https://github.com/strogonoff/geolexica-database).

2. A different convention can be considered for adoption, but it’s best done as part of next larger migration (e.g., migration to per-language subregistries). Until then, unexpected data changes could negatively affect software stability.

**Comment by skalee** [comment]
Sorry for late answer on that.

> This depends on whether we consider this repository native application storage or export artifact.

IMHO it is more than just some kind of storage.

Firstly, these concept YAML can be downloaded directly from glossary sites as machine-readable representation of concepts, so they're kinda published artifacts. For example, see the following screenshot from [IEV 103-08-14](https://iev-staging.geolexica.org/concepts/103-08-14). Among the download links you can see YAML (which is exactly in a format used in this repo) and JSON (which has the same structure, just in JSON):

| ![Zrzut ekranu 2020-07-02 o 03 01 57](https://user-images.githubusercontent.com/154287/86304900-93007200-bc10-11ea-865c-e75db7d6e456.png) |
| ---- |

Secondly, there are plans to [create a Ruby gem](https://github.com/glossarist/iev-data/issues/15) for reading and writing these concept YAMLs. So again, they're bit more than internal format for a desktop app. Though I have to add that I'm not sure if this gem is still desired, given the fact that we are moving away from spreadsheet parsing. And for sure I have few more bugfixes to do before I have time to implement that gem.

In my understanding, this is not just an internal storage format, but rather an API format, which is meant to be consumed by third-party applications. If I'm correct, then we should stabilize the format at some point, and adopt a consistent naming convention. Either underscored or camel-cased, but not mixed. From my point of view both are fine.

Perhaps we should ask @ronaldtse for clarification. CC @strogonoff.

**Comment by ronaldtse** [comment]
> Secondly, there are plans to create a Ruby gem for reading and writing these concept YAMLs. So again, they're bit more than internal format for a desktop app. Though I have to add that I'm not sure if this gem is still desired, given the fact that we are moving away from spreadsheet parsing. And for sure I have few more bugfixes to do before I have time to implement that gem.

Yes we still need to create this Ruby gem. We recently have a task to extract all terms out from a data set, and those terms have to be ingested into Metanorma (for publication), and then extracted out again into a machine-readable format (for reuse), which I believe is best to be in this gem-created format, so that Glossarist can also be used to manage them. 

In the future, Metanorma can also read in terms defined by this YAML format, using this gem.

> In my understanding, this is not just an internal storage format, but rather an API format, which is meant to be consumed by third-party applications. If I'm correct, then we should stabilize the format at some point, and adopt a consistent naming convention. Either underscored or camel-cased, but not mixed. From my point of view both are fine.

Agree.
